cabal-install/src/Distribution/Client/RebuildMonad.hs

```diff
@@ -123,39 +127,61 @@ rerunIfChanged
   -> Rebuild b
   -> Rebuild b
 rerunIfChanged verbosity monitor key action = do
+  -- rerunIfChanged is implemented in terms of rerunConcurrentlyIfChanged, but
+  -- nothing concurrent will happen since the list of concurrent actions has a
+  -- single value that will be waited for alone.
```

Comment:
This sounds smart at first, but after thinking more: is it a good idea to put two concerns (checking monitors and doing the concurrent map) at once? It'd be better if instead of `rerunIfChanged` depending on `rerunConcIfChanged` it was the other way round: the latter would only implement the concurrent map and call the former when it gets to the point of checking a monitor for one value. But maybe this is not possible?

Reply:
I think I tried that originally but it didn't work / the result was not satisfactory.